server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsServiceImpl.java

```diff
@@ -126,10 +188,333 @@ protected String prepareShellCmd(String cmdType, String ipAddress, String option
         }
     }
 
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_SYSTEM_VM_DIAGNOSTICS, eventDescription = "getting diagnostics files on system vm", async = true)
```

Comment:
in my understanding this API could be run on hosts as well as (system)VMs. has something changed?

Reply:
from what I see Daan, this is purely for systemvms.